docs/egress.md

```diff
@@ -0,0 +1,297 @@
+- [The ExternalIPPool resource](#the-externalippool-resource)
+  - [IPRanges](#ipranges)
+  - [nodeSelector](#nodeselector)
+- [Usage examples](#usage-examples)
+  - [Configuring high available Egress](#configuring-high-available-egress)
+  - [Configuring static Egress](#configuring-static-egress)
+- [Limitations](#limitations)
+<!-- /toc -->
+
+## What is Egress?
+
+`Egress` is a CRD API that manages external access from the Pods in a cluster.
+It supports specifying which egress (SNAT) IP the traffic from the selected Pods
+to the external network should use. When a selected Pod accesses the external
+network, the egress traffic will be tunneled to the Node that hosts the egress
+IP if it's different from the Node that the Pod runs on and will be SNATed to
+the egress IP when leaving that Node.
+
+You may be interested in using this capability if any of the below are true:
+
+- A consistent IP address is desired when specific Pods connect to services
+  outside of the cluster, for source tracing in audit logs, or whitelisting
+  by source IP in external firewall, etc.
+
+- You want to limit outgoing external connections leave the cluster via certain
+  Nodes, for security controls, or due to the network topology restriction.
+
+This guide demonstrates how to configure `Egress` to achieve the above result.
+
+## Prerequisites
+
+Egress is introduced in v1.0 as an alpha feature. As with other alpha features,
+a feature gate `Egress` must be enabled on the antrea-controller and
+antrea-agent for the feature to work. The following options in the
+`antrea-config` ConfigMap need to be set:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    featureGates:
+      Egress: true
+  antrea-controller.conf: |
+    featureGates:
+      Egress: true
+```
+
+## The Egress resource
+
+A typical Egress resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-prod-web
+spec:
+  appliedTo:
+    namespaceSelector:
+      matchLabels:
+        env: prod
+    podSelector:
+      matchLabels:
+        role: web
+  egressIP: 10.10.0.8
+  externalIPPool: prod-external-ip-pool
+status:
+  egressNode: node01
+```
+
+### AppliedTo
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+Egress applies to. Pods can be selected cluster-wide using `podSelector`. If set
+with a `namespaceSelector`, all Pods from Namespaces selected by the
+`namespaceSelector` will be selected. Specific Pods from specific Namespaces can
+be selected by providing both a `podSelector` and a `namespaceSelector`. Empty
+`appliedTo` selects nothing. The field is mandatory.
+
+### EgressIP
+
+The `egressIP` field specifies the egress (SNAT) IP the traffic from the
+selected Pods to the external network should use. **The IP must be reachable
+from all Nodes.** The IP can be specified when creating the Egress. Starting
```

Comment:
**The IP must be reachable from all Nodes.**

I wonder if this is really necessary. If I have a cluster with Nodes spread over multiple Node subnets, it should be ok to partition my egress IPs over the different subnets, and each egress IP only needs to be reachable from within the subnet in which it is used. I would configure the `nodeSelector` in my IP pools to only select Nodes within that subnet. But it's probably an unusual scenario and it's easier to frame the requirement as "The IP must be reachable from all Nodes".

Reply:
The IP must be reachable from all Nodes because we use the egress IPs as tunnel destination (to "encap" the snat IP the traffic should use). As any Pod on any Node can use the egress IPs, the source Node must be able to reach the egress IP, otherwise the tunnel traffic would be dropped.
https://github.com/antrea-io/antrea/blob/main/docs/design/ovs-pipeline.md#snattable-71 has details about the SNAT IP as Tunnel destination IP.